### ISSUE TITLE
Allow Query to update integer/double column values with binary operators

### ISSUE THREAD

**Issue by joeconwaystk** [stablekernel/aqueduct#446]
For example, update an account's balance by adding 100 to it - without first having to fetch the current value.

**Comment by joeconwaystk** [comment]
Whatever solution is chosen, it must support addition, subtraction, multiplication and division at minimum. It is preferable that the same solution could be used for other types of columns, e.g, adding a value to an array, or inserting a key in a `Document`.

**Comment by joeconwaystk** [comment]
This feature should use the property selector syntax, and return an object like `QueryExpression` with instance methods like `add`, `divideBy`, etc. When Dart adds generic overrides, this behavior can be extended to provide a different object/instance methods depending on the property being modified. An example looks like:

```dart
final q = new Query<Employee>()
  ..modify((e) => e.salary).add(1000);
```

A query that modifies and sets the same property (via `values`) should throw an error.

**Comment by RedOneKH** [comment]
so how to translate this ?
`UPDATE Employee SET salary = salary + 100 `

**Comment by joeconwaystk** [comment]
You'll need to execute that query manually:

```dart
await context.store.execute("UPDATE _employee SET salary = salary + 100");
```

**Comment by RedOneKH** [comment]
so this is the only solution right now ?

**Comment by joeconwaystk** [comment]
Correct. The intent of the ORM is not to exactly fit SQL, but to provide convenience for operations as time allows.

**Comment by RedOneKH** [comment]
Okey thank you !!!
